### PR TITLE
CalendarContextProvider - use with pure functions

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Move `setDate` and `setDisabled` back to Provider and removing the Presenter's relevant tests

### CODE REVIEW

File: src/expandableCalendar/Context/Provider.tsx [hunk 1]
```diff
@@ -65,20 +63,19 @@ const CalendarProvider = (props: CalendarContextProviderProps) => {
...
+  const today = useRef(getTodayFormatted());
   const wrapperStyle = useMemo(() => {
     return [style.current.contextWrapper, propsStyle];
   }, [style, propsStyle]);
 
+  const [prevDate, setPrevDate] = useState(date);
+  const [currentDate, setCurrentDate] = useState(date);
+  const [updateSource, setUpdateSource] = useState(UpdateSources.CALENDAR_INIT);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [buttonIcon, setButtonIcon] = useState(getButtonIcon(date, showTodayButton));
```
Comment (ethanshar) [inline src/expandableCalendar/Context/Provider.tsx:77]: Move useState's before useMemo's

File: src/expandableCalendar/Context/Provider.tsx [hunk 2]
```diff
@@ -89,24 +86,27 @@ const CalendarProvider = (props: CalendarContextProviderProps) => {
...
+  /** Context */
+
   const _setDate = (date: string, updateSource: UpdateSources) => {
-    const updateState = (buttonIcon: number) => {
-      setPrevDate(currentDate);
-      setCurrentDate(date);
-      setUpdateSource(updateSource);
-      setButtonIcon(buttonIcon);
-    };
+    setPrevDate(currentDate);
```
Comment (ethanshar) [inline src/expandableCalendar/Context/Provider.tsx:92]: I wonder of these calls trigger multiple renders

**Review by ethanshar — APPROVED** [review]

File: src/expandableCalendar/Context/Provider.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/expandableCalendar/Context/Provider.tsx:92]: When I log the provider, I see one call after `setDate`.

File: src/expandableCalendar/Context/Provider.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/expandableCalendar/Context/Provider.tsx:77]: Done

**Pull request merged**